tsl/src/continuous_aggs/invalidation_threshold.c

```diff
@@ -284,3 +251,63 @@ invalidation_threshold_compute(const ContinuousAgg *cagg, const InternalTimeRang
 
 	return refresh_window->end;
 }
+
+void
+invalidation_threshold_insert(const ContinuousAgg *cagg, int64 threshold, bool threshold_isnull)
+{
+	bool found = false;
+	ScanKeyData scankey[1];
+	Catalog *catalog = ts_catalog_get();
+	ScannerCtx scanctx = {
+		.table = catalog_get_table_id(catalog, CONTINUOUS_AGGS_INVALIDATION_THRESHOLD),
+		.index =
+			catalog_get_index(catalog, CONTINUOUS_AGGS_INVALIDATION_THRESHOLD, BGW_JOB_PKEY_IDX),
+		.nkeys = 1,
+		.scankey = scankey,
+		.limit = 1,
+		.lockmode = ShareUpdateExclusiveLock,
+		.scandirection = ForwardScanDirection,
+		.result_mctx = CurrentMemoryContext,
+		.flags = SCANNER_F_KEEPLOCK,
+	};
+
+	ScanKeyInit(&scankey[0],
+				Anum_continuous_aggs_invalidation_threshold_hypertable_id,
+				BTEqualStrategyNumber,
+				F_INT4EQ,
+				Int32GetDatum(cagg->data.raw_hypertable_id));
+
+	found = ts_scanner_scan_one(&scanctx, false, "invalidation threshold");
+
+	DEBUG_WAITPOINT("invalidation_threshold_insert");
+
+	if (!found)
```

Comment:
Hmmm... seems this is another different error (I guess)... didn't hit it before

Reply:
The way  `invalidation_threshold_insert` seems to be used (only at cagg creation) we never expect a conflict (e.g., a value already exists). Therefore, I am wondering if we should assert or raise on error in case a threshold is found during insert (e.g., `found==true`).